lib/uplinkc/scope.go

```diff
@@ -32,3 +39,56 @@ func (parent *scope) child() scope {
 	ctx, cancel := context.WithCancel(parent.ctx)
 	return scope{ctx, cancel}
 }
+
+//export restrict_scope
+// restricts a given scope with the provided caveat and encryptionRestrictions
+func restrict_scope(scopeRef C.ScopeRef, caveatRef C.CaveatRef, restrictions **C.EncryptionRestriction, cerr **C.char) C.ScopeRef {
```

Comment:
this needs to be placed in `acces_scope.go`.

Reply:
Switch function to access_scope